### ISSUE TITLE
BeNull is not available for Nullable int

### ISSUE THREAD

**Issue by MNF** [dennisdoomen/FluentAssertions#495]
When I tried to check if int? variable is null, I've got compiler error:

Error	CS1061	'NullableNumericAssertions<int>' does not contain a definition for 'BeNull' and no extension method 'BeNull' accepting a first argument of type 'NullableNumericAssertions<int>' could be found (are you missing a using directive or an assembly reference?).

Example code:
```
  [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        public void TestMethod1()
        {
            int? nullableInt = null;
            Assert.IsNull(nullableInt);
            nullableInt.Should().BeNull();
        }
    }
```

**Comment by dennisdoomen** [comment]
No, for nullable types, you're supposed to use `Should().HaveValue` and `Should().NotHaveValue`

**Comment by somewhatabstract** [comment]
Is there a solid reason why we cannot have `BeNull` and `NotBeNull` as well?

**Comment by dennisdoomen** [comment]
No, not really. It's just that I preferred a more functional name.

**Comment by MNF** [comment]
I expected for nullable fields to have standard BeNull/NotBeNull, adding these methods will be nice.

**Comment by dennisdoomen** [comment]
Sure. I'd welcome a PR ;-)

**Comment by dennisdoomen** [comment]
Released in 4.18.0
